stylus-proc/src/macros/public/types.rs

```diff
@@ -77,11 +94,21 @@ impl PublicImpl {
             .collect::<Vec<_>>();
         let inheritance_routes = self.inheritance_routes();
 
+        // Prepares the statement that invokes a fallback function if it exists.
+        // If it doesn't exist, we rely on the inherited fallback function.
+        // Also, a fallback could be defined in a trait our type implements, so we pass
+        // the trait_ option to the token builder method below. For instance, if
+        // our type `MyContract` implements `MyTrait`, and `MyTrait` has a fallback function, while
+        // MyContract does NOT have a fallback function, we need to call the fallback function from `MyTrait`.
+        // We need to call `<MyContract as MyTrait>::fallback` instead of `MyContract::fallback`.
+        let fallback_tokens_builder = FallbackTokenBuilder {
+            trait_: self.trait_.clone(),
+        };
```

Comment:
Could this be a lambda function instead? I think having the lambda definition here would make the code a bit simpler to understand.